pkg/services/ngalert/notifier/channels/webhook.go

```diff
@@ -152,12 +177,18 @@ func (wn *WebhookNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool
 		return false, err
 	}
 
+	headers := make(map[string]string, 1)
```

Comment:
I don't believe 1 is required here as 1 is the default size?

```suggestion
	headers := make(map[string]string)
```

Reply:
Can you link to a reference where this is stated? I thought it was 0.

Reply:
`1` is a hint, and for stack allocated maps this is `0` for hints up to `8`. I understand that is here https://github.com/golang/go/blob/master/src/runtime/map.go#L299-L337.

Reply:
`1` is a hint, and for stack allocated maps I understand the size is `0` for hints at least up to `8`. For hints greater than 8 the load factor is taken into account when choosing the initial size. The code I am referring to can be found here https://github.com/golang/go/blob/master/src/runtime/map.go#L299-L337.

Reply:
`1` is a hint, and for stack allocated maps I understand the size is `0` for hints at least up to `8`. For hints greater than `8` the load factor is taken into account when choosing the initial size. The code I am referring to can be found here https://github.com/golang/go/blob/master/src/runtime/map.go#L299-L337.

Reply:
`1` is a hint, and for stack allocated maps I understand the size is `0` for hints at least up to `8`. For hints larger than `8` the load factor is taken into account when choosing the initial size. The code I am referring to can be found here https://github.com/golang/go/blob/master/src/runtime/map.go#L299-L337.

Reply:
Sorry! I wasn't sure I understood correctly what you've written here, so I decided to do some digging. In particular:

> size is 0 for hints at least up to 8.

[this source](https://levelup.gitconnected.com/memory-allocation-and-performance-in-golang-maps-b267b5ad9217) does a decent explanation of expanding on this concept (TIL!) - see the links in "map structure" section.

My takeaway is that a map initialized with no hint will have one bucket with eight elements by default - effectively making a default size of 8 key/value pairs. 

> A map in Go is a hash table that stores its key/value pairs into buckets. Each bucket is an array that holds up to 8 entries. The default number of buckets is 1.

Thanks!